src/main/java/gregtech/api/unification/stack/ItemMaterialInfo.java

```diff
@@ -18,6 +18,13 @@ public ItemMaterialInfo(List<MaterialStack> materials) {
         this.materials.addAll(materials);
     }
 
+    public ItemMaterialInfo(List<ItemMaterialInfo> infos, MaterialStack... materials) {
+        for (ItemMaterialInfo info : infos) {
+            this.materials.addAll(info.getMaterials());
+        }
+        this.materials.addAll(Arrays.asList(materials));
```

Comment:
You can use `Collections.addAll(this.materials, materials)` here instead, which skips needing to create another list from the array first.